src/ui/RelevanceInspector/MetaDataTable.ts

```diff
@@ -0,0 +1,220 @@
+import { TableBuilder, ITableDataSource } from './TableBuilder';
+import { map, each, find } from 'underscore';
+import agGridModule = require('ag-grid/main');
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { $$, Dom } from '../../utils/Dom';
+import { StringUtils } from '../../utils/StringUtils';
+import { DateUtils } from '../../utils/DateUtils';
+import { IResultsComponentBindings } from '../Base/ResultsComponentBindings';
+import { IRelevanceInspectorTab } from './RelevanceInspector';
+import { TextInput } from '../FormWidgets/TextInput';
+import { IFieldDescription } from '../../rest/FieldDescription';
+
+export class MetaDataTable implements IRelevanceInspectorTab {
+  public gridOptions: agGridModule.GridOptions;
+
+  constructor(public results: IQueryResult[], public bindings: IComponentBindings) {}
+
+  public async build(): Promise<Dom> {
+    if (!this.bindings.queryController) {
+      return;
+    }
+
+    const fieldsDescription = await this.bindings.queryController.getEndpoint().listFields();
+
+    const container = $$('div', {
+      className: 'metadata-table'
+    });
+
+    const builders = this.results.map(async result => {
+      const nestedContainer = $$('div');
+      container.append(nestedContainer.el);
+
+      const nestedAgGrid = $$('div', {
+        className: 'ag-theme-fresh'
+      });
+
+      nestedContainer.append(nestedAgGrid.el);
+
+      const thumbnail = TableBuilder.thumbnailCell(result, this.bindings as IResultsComponentBindings);
+      const fields: Record<string, ITableDataSource> = {};
+
+      fields[`Fields Values`] = {
+        content: { result, fieldsDescription },
+        cellRenderer: FieldValuesRenderer,
+        width: 1000,
+        getQuickFilterText: (params: agGridModule.GetQuickFilterTextParams) => {
+          const allValues = map(params.value.result.raw, val => val.toString());
+          return Object.keys(params.value.result.raw)
```

Comment:
It's taking an object like this : 

```
{
 foo : 'bar',
 bazz : 1234,
 randomKey: true
}
```

Takes all the values, and transform them to string in an array : 

```
['bar', '1234', 'true']
```

Then concat the keys : 
```
['bar', '1234', 'true', 'foo', 'bazzz',  'randomKey']
```

So that when you search the table, it searches for both the key and values for that object (the metadata on each result).